src/optimizer/rule/IndexScanRule.h

```diff
@@ -0,0 +1,152 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+#ifndef OPTIMIZER_INDEXSCANRULE_H_
+#define OPTIMIZER_INDEXSCANRULE_H_
+
+#include "optimizer/OptRule.h"
+#include "planner/Query.h"
+#include "optimizer/OptimizerUtils.h"
+
+namespace nebula {
+namespace graph {
+class IndexScan;
+}   // namespace graph
+namespace opt {
+
+using graph::IndexScan;
+using graph::OptimizerUtils;
+using graph::PlanNode;
+using graph::QueryContext;
+using storage::cpp2::IndexQueryContext;
+using storage::cpp2::IndexColumnHint;
+using BVO = graph::OptimizerUtils::BoundValueOperator;
+using IndexItem = std::shared_ptr<meta::cpp2::IndexItem>;
+using IndexQueryCtx = std::unique_ptr<std::vector<IndexQueryContext>>;
+using FilterItem = std::tuple<std::string, RelationalExpression::Kind, Value>;
```

Comment:
Better add some comments for this `filterItem`,  or make it a struct.

Reply:
> Better add some comments for this `filterItem`, or make it a struct.

I will second this. My brain was screwed while figuring out the exact meaning of each field of the tuple, especially at `std::get<>`

Reply:
> > Better add some comments for this `filterItem`, or make it a struct.
> 
> I will second this. My brain was screwed while figuring out the exact meaning of each field of the tuple, especially at `std::get<>`

I take the advice . let me improve it.